src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -39,6 +39,15 @@ import { relativeToAbsolute } from '../../nav_links/to_nav_link';
 export const isModifiedOrPrevented = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
   event.metaKey || event.altKey || event.ctrlKey || event.shiftKey || event.defaultPrevented;
 
+// Future: should make this function more generic
+export const isActiveNavLink = (currentId: string | undefined, targetId: string): boolean => {
+  if (currentId === 'discover' || currentId === 'data-explorer') {
+    return ['discover', 'data-explorer'].includes(targetId);
+  } else {
+    return currentId === targetId;
+  }
+};
```

Comment:
@Willie-The-Lord I also agree that this is a temporary solution, but one worth moving ahead with while we do more investigation into how to improve `core.application.register()` to be more flexible for specifying app link properties (perhaps `navLinkStatus` needs to be expanded to optionally take a function, or we need to create a registration point for active app link similar to how we handle breadcrumbs).

But in the meantime, I think we could at least separate out the special cases, and only check them as a fallback, something like this (note that this is just pseudo-code for illustration):

```suggestion
// Some apps share views with other apps that should also mark the nav link as active
// TODO: replace hard-coded values with a registration function, so that apps can control active nav links similar to breadcrumbs
const aliasedApps = {
    discover: ['data-explorer'],
    visualize: ['vis-builder'],
}

export const isActiveNavLink = (appId: string | undefined, linkId: string): boolean => (
  appId === linkId || aliasedApps[linkId]?.includes(appId);
);
```

Reply:
Thanks for the insight! You're right, this situation does seem a lot like dealing with breadcrumbs – that idea didn't cross my mind before! I believe the current fix is suitable for now, but I would be really interested in expanding the functionality in the future. Working on this improvement would be a great way for me to better understand how `core.application` functions.